src/__experimental__/components/select/select.stories.js

```diff
@@ -13,34 +13,91 @@ const multiSelectStore = new Store({
   value: []
 });
 
+const commonKnobs = (store) => {
+  const filterable = boolean('filterable', Select.defaultProps.filterable);
+  const typeAhead = filterable && boolean('typeAhead', Select.defaultProps.typeAhead);
+
+  return {
+    disabled: boolean('disabled', false),
+    errorMessage: text('errorMessage'),
+    infoMessage: text('infoMessage'),
+    label: text('label'),
+    labelAlign: select('labelAlign', OptionsHelper.alignBinary),
+    labelInline: boolean('labelInline', false),
+    onChange: (ev) => { store.set({ value: ev.target.value }); },
+    placeholder: text('placeholder'),
+    readOnly: boolean('readOnly', false),
+    size: select('size', OptionsHelper.sizesRestricted),
+    warningMessage: text('warningMessage'),
+    filterable,
+    typeAhead
+  };
+};
+
+const selectOptions = [
```

Comment:
isn't it possible to map through all the values the code is not repeated?

Reply:
Yes I believe so. I've refactored this in 54ecf5e :thumbsup: